src/Symfony/Component/Serializer/Normalizer/ObjectNormalizer.php

```diff
@@ -159,4 +129,55 @@ public function denormalize($data, $class, $format = null, array $context = arra
 
         return $object;
     }
+
+    /**
+     * Gets and caches attributes for this class and context.
+     *
+     * @param object $object
+     * @param array  $context
+     *
+     * @return array
+     */
+    private function getAttributes($object, array $context)
+    {
+        $key = sprintf('%s-%s', get_class($object), serialize($context));
+
+        if (isset(self::$attributesCache[$key])) {
+            return self::$attributesCache[$key];
+        }
+
+        if (self::$attributesCache[$key] = $this->getAllowedAttributes($object, $context, true)) {
+            return self::$attributesCache[$key];
+        }
+
+        // If not using groups, detect manually
+        $attributes = array();
+
+        // methods
+        $reflClass = new \ReflectionClass($object);
+        foreach ($reflClass->getMethods(\ReflectionMethod::IS_PUBLIC) as $reflMethod) {
+            if (
+                !$reflMethod->isConstructor() &&
+                !$reflMethod->isDestructor() &&
+                0 === $reflMethod->getNumberOfRequiredParameters()
+            ) {
+                $name = $reflMethod->getName();
+
+                if (strpos($name, 'get') === 0 || strpos($name, 'has') === 0) {
+                    // getters and hassers
+                    $attributes[lcfirst(substr($name, 3))] = true;
+                } elseif (strpos($name, 'is') === 0) {
+                    // issers
+                    $attributes[lcfirst(substr($name, 2))] = true;
+                }
```

Comment:
Aren't the constructor checks redundant here? They would never get through the has/is/get checks, or is it done simply because of performance reasons?

Personally I would like to see a guard clause here for readability:

```php
if ($reflMethod->getNumberOfRequiredParameters() > 0) {
    continue;
}
// also prevents a lot of is checks as destructor is hardly ever implemented and constructor only ever once
```

Another option would be to omit the number of required attributes at this point and only check it when needed.
```php
$attributes[lcfirst(substr($name, 3))] = 0 === $reflMethod->getNumberOfRequiredParameters();
```
This would limit the checks even more, but I'm not sure if calling the reflection method is slower or relying on `getName()` and `strpos()`

_If it comes down to preferences, feel free to ignore if you disagree_